Program.fs

```diff
@@ -183,7 +186,13 @@ let getSatisfyingPackageVersion (nodeManager: NodeManager) (pkg : NpmDependency)
                 |> CreateProcess.ensureExitCode
                 |> Proc.run
 
-            Decode.unsafeFromString (Decode.field "data" (Decode.list Decode.string)) res.Result.Output
+            // Yarn returns ExitCode = 0 even if the packge is not found,
+            // so we need to check using StrErr channel
+            if String.isNullOrEmpty res.Result.Error then
+                Decode.unsafeFromString (Decode.field "data" (Decode.list Decode.string)) res.Result.Output
+            else
+                let errorMessage = Decode.unsafeFromString (Decode.field "data" Decode.string) res.Result.Error
+                failwith errorMessage
```

Comment:
> and we could forget to try...catch it somewhere.

Indeed that's why It tried to return `Result<string list, string>` at first but it was messy too.

> Write a specialized function that checks whether a package exists and let it run in the beginning of the validation process, if a package does not exists, then filter it out (after logging it of course) and let the rest of the packages go through the validation process to get version metadata etc.

This will double the execution time of Femto. 

It seems to take around 1.5 seconds for a non-existant package and 0.7 seconds for a real package (and yes I have a good connection in theory ^^)

But your solution gives me an idea, we could fetch all the packages info at the beginning filter out from this list the non-existant ones and they process normally.

Like that we fetch them only once. I will give it a try it and share the result with you.

Reply:
Sounds like a good idea, having some static `ResizeArray` somewhere in the app for the packages.

Although I personally don't mind waiting a few extra seconds in Femto does what it should do ;)

Reply:
>  Although I personally don't mind waiting a few extra seconds if Femto does what it should do ;)

Yes, but having 10 packages means 7 seconds to execute.

Reply:
If we do 1 pass only.